### DISCUSSION TITLE
New feature to Save Lists/Groups and recall them from the Search Bar!?

### DISCUSSION THREAD

**Discussion by ahmedfarrag17** [espanso/espanso#1250] [Feature requests and Ideas]
Hello.
I would like to see this feature in Espanso. It would be extremely useful.

It is available in Ditto Clipboard Manager software but Ditto is only for Windows and not intuitive.

Feature:
Being able to Pre-save Groups (folders) (and sub-groups) of strings/syntax/Text, and give a Label (indication/name) to each string so we can recall them easily from the Espanso Search Bar.

(Also, have the possibility to add "After Paste" actions, and "Before Copy actions". See example below).

**Practical Example**:
Let's say I type in a document and I insert a lot of LaTeX maths equations. I don't remember them. So in Espanso I create a Group called "LaTeX". In which I save \sqrt and give it the name "Square root". (And many others strings like this). 
So after that I open Espanso Search Bar 🔍 and search "square r....". I find it and once i Click Enter, Espanso copies and pastes it in the current mouse position (in my document).

**After paste Example**:
For specific groups (or sub-groups) when I copy elements from them I want Espanso to execute an action immediately after pasting. (For ex. Do automatically "Alt+X" after pasting Unicode numbers (ex. 2303) to convert them into symbols).

**Comment by federico-terzi** [comment]
Hey @ahmedfarrag17,

Thank you for the feedback! This seems related to #288, I'll mark it as a feature request

Your "After paste" example will definitely be possible once Macros #556 are implemented :)

Cheers :)

**Comment by ahmedfarrag17** [comment]
@federico-terzi Thank you! I will rely on Ditto for now, unfortunately. but I will be waiting for this amazing feature to be implemented. Espanso is a great tool and I use it as a text expander, especially for Emojis which is really cool :).
I appreciate your hard work. Thank you again! 

Update: 
I discovered "Choice Extension" which is probably what I was looking for. I will try it! 
Thank you, Fede! 😆

**Comment by federico-terzi** [comment]
Sounds good! Thanks :)
